stl/inc/ranges

```diff
@@ -6842,6 +6843,472 @@ namespace ranges {
+        requires (view<_ViewTypes> && ...) && (sizeof...(_ViewTypes) > 0)
+    class zip_view : public view_interface<zip_view<_ViewTypes...>> {
+        // clang-format on
+    private:
+        /* [[no_unique_address]] */ tuple<_ViewTypes...> _Views;
+
+        template <bool _IsConst>
+        struct _Category_base {};
+
+        template <bool _IsConst>
+            requires _All_forward<_IsConst, _ViewTypes...>
+        struct _Category_base<_IsConst> {
+            using iterator_category = input_iterator_tag;
+        };
+
+        template <bool _IsConst>
+        class _Iterator : public _Category_base<_IsConst> {
+        private:
+            /* [[no_unique_address]] */ tuple<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>...> _Current;
+
+            friend zip_view;
+
+            constexpr explicit _Iterator(tuple<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>...> _Current_) noexcept(
+                is_nothrow_move_constructible_v<
+                    tuple<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>...>>) // strengthened
+                : _Current(_STD move(_Current_)) {}
+
+        public:
+            using iterator_concept = conditional_t<_All_random_access<_IsConst, _ViewTypes...>,
+                random_access_iterator_tag,
+                conditional_t<_All_bidirectional<_IsConst, _ViewTypes...>, bidirectional_iterator_tag,
+                    conditional_t<_All_forward<_IsConst, _ViewTypes...>, forward_iterator_tag, input_iterator_tag>>>;
+
+            using value_type      = tuple<range_value_t<_Maybe_const<_IsConst, _ViewTypes>>...>;
+            using difference_type = common_type_t<range_difference_t<_Maybe_const<_IsConst, _ViewTypes>>...>;
+
+            _Iterator() = default;
+
+            constexpr _Iterator(_Iterator<!_IsConst> _Rhs) noexcept(
+                (is_nothrow_convertible_v<iterator_t<_ViewTypes>,
+                     iterator_t<_Maybe_const<!_IsConst, _ViewTypes>>> && ...)) // strengthened
+                // clang-format off
+                requires(_IsConst &&
+            (convertible_to<iterator_t<_ViewTypes>, iterator_t<_Maybe_const<_IsConst, _ViewTypes>>> && ...))
+                // clang-format on
+                : _Current(_STD move(_Rhs._Current)) {}
+
+            _NODISCARD constexpr auto operator*() const
+                noexcept((noexcept(*(_STD declval<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>()))
+                          && ...)) /* strengthened */ {
+                return _Tuple_transform([](auto& _Itr) -> decltype(auto) { return *_Itr; }, _Current);
+            }
+
+            constexpr _Iterator& operator++() noexcept(
+                noexcept(_Tuple_for_each([](auto& _Itr) { ++_Itr; }, _Current))) /* strengthened */ {
+                _Tuple_for_each([](auto& _Itr) { ++_Itr; }, _Current);
+                return *this;
+            }
+
+            constexpr void operator++(int) noexcept(noexcept(++(_STD declval<_Iterator>()))) /* strengthened */ {

```

Comment:
I believe we should be able to say `noexcept(noexcept(++*this))` here. Occurs below, also for decrement.

Reply:
I'm not so sure that works, actually. When I do it like that, I get MSVC compilation error `C2355: 'this': can only be referenced inside non-static member functions or non-static data member initializers`. I'm going to keep the `_STD declval<_Iterator>()` version for now, but let me know if I am missing something.